lib/routing.js

```diff
@@ -62,7 +65,20 @@ export const deserializeState = serializedState => {
   return JSON.parse(decodeURIComponent(stateString))
 }
 
-const getQueryStringObject = query => {
+export const getRouteState = location => {
+  const { query, pathname } = url.parse(location, true)
+  const state = getQueryStringState(query)
+  const path = escapeHtml(pathname.split('/').pop())
+  const gistUri = path.length >= 19 && path.indexOf('.') === -1 && path
+
+  return {
+    gistUri,
```

Comment:
Is `gistUri` the correct word for this? I am thinking it should be like `gistPath` or `gistId`